apps/dashboard/app/models/files.rb

```diff
@@ -144,14 +144,21 @@ def self.mime_type_by_extension(path)
   end
 
   def self.username(uid)
-    @username_for_ids ||= Hash.new do |h, key|
-      h[key] = begin
-        Etc.getpwuid(uid).name
```

Comment:
Actually ya know what, this memorization patter does work. Only this needs to be `Etc.getpwuid(key).name` (using the key) not the field `uid`.